src/e3/electrolyt/plan.py

```diff
@@ -21,6 +23,57 @@ class PlanError(E3Error):
     pass
 
 
+class ConditionalConstant:
+    """Allow defining constant in the plan that can depend on external values.
+
+    Note that ConditionalConstant are meant to enable/disable part of the
+    plan but not to modify the plan completely. In particular it is not
+    recommended to write things such as:
+
+        if my_const:
+            anod_build("my_component")
+
+    In that case tools parsing the plan won't be able to detect potential issues.
+    Instead use:
+
+        with defaults(enabled="my_const"):
+             anod_build("my_component")
+    """
+
+    RECORD: List[ConditionalConstant] = []
```

Comment:
I think it would be better to introduce ConstantSet class and move RECORD and shuffle in it

Reply:
Ok, I'll try.